### DISCUSSION TITLE
Does Deepgram TTS Support Real-time Input?

### DISCUSSION THREAD

**Discussion by suptejas** [deepgram/community#704] [General help] [answered]
My ideal workflow is the follows:

LLM generates and streams output -> streamed in real-time to Deepgram TTS API -> get real-time output stream for speech.

Essentially the speech is heard as the LLM output is generated.

Is this currently possible today - either with native Deepgram support or a "hacky" approach?

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jpvajda** [comment, marked as answer]
@suptejas  Currently our TTS API offers a way to start streaming the audio output on arrival of the first byte, [this Doc ](https://developers.deepgram.com/docs/streaming-the-audio-output)has more details on that. 

We are currently looking into offering a Streaming API for TTS that can handle input streaming but I don't have an ETA of when that feature will be available. 

If you want join our Discord Channel we keep the community up to date on new releases.  https://dpgr.am/discord
